internal/app/machined/pkg/controllers/time/sync.go

```diff
@@ -0,0 +1,157 @@
+func (ctrl *SyncController) Name() string {
+	return "time.SyncController"
+}
+
+// ManagedResources implements controller.Controller interface.
+func (ctrl *SyncController) ManagedResources() (resource.Namespace, resource.Type) {
+	return v1alpha1.NamespaceName, time.StatusType
+}
+
+// Run implements controller.Controller interface.
+//
+//nolint: gocyclo,cyclop
+func (ctrl *SyncController) Run(ctx context.Context, r controller.Runtime, logger *log.Logger) error {
+	if err := r.UpdateDependencies([]controller.Dependency{
+		{
+			Namespace: config.NamespaceName,
+			Type:      config.MachineConfigType,
+			ID:        pointer.ToString(config.V1Alpha1ID),
+			Kind:      controller.DependencyWeak,
+		},
+	}); err != nil {
+		return fmt.Errorf("error setting up dependencies: %w", err)
+	}
+
+	var (
+		syncCtx       context.Context
+		syncCtxCancel context.CancelFunc
+		syncWg        sync.WaitGroup
+
+		syncCh  <-chan struct{}
+		epochCh <-chan struct{}
+		syncer  *ntp.Syncer
+
+		timeSynced bool
+		epoch      int
+	)
+
+	defer func() {
+		if syncer != nil {
+			syncCtxCancel()
+
+			syncWg.Wait()
+		}
+	}()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-r.EventCh():
+		case <-syncCh:
+			syncCh = nil
+			timeSynced = true
+		case <-epochCh:
+			epoch++
+		}
+
+		cfg, err := r.Get(ctx, resource.NewMetadata(config.NamespaceName, config.MachineConfigType, config.V1Alpha1ID, resource.VersionUndefined))
+		if err != nil {
+			if !state.IsNotFoundError(err) {
+				return fmt.Errorf("error getting config: %w", err)
+			}
+		}
+
+		syncDisabled := false
+
+		if ctrl.V1Alpha1State.Platform().Mode() == v1alpha1runtime.ModeContainer {
+			syncDisabled = true
+		} else if cfg != nil && cfg.(*config.MachineConfig).Config().Machine().Time().Disabled() {
+			syncDisabled = true
+		}
+
+		timeServers := []string{constants.DefaultNTPServer}
+		if cfg != nil {
+			timeServers = cfg.(*config.MachineConfig).Config().Machine().Time().Servers()
+		}
+
+		if syncDisabled && syncer != nil {
+			// stop syncing
+			syncCtxCancel()
+
+			syncWg.Wait()
+
+			syncer = nil
+			syncCh = nil
+			epochCh = nil
+		} else if !syncDisabled && syncer == nil {
+			// start syncing
+			syncer = ntp.NewSyncer(logger, timeServers)
+			syncCh = syncer.Synced()
+			epochCh = syncer.EpochChange()
+
+			timeSynced = false
+
+			syncCtx, syncCtxCancel = context.WithCancel(ctx) // nolint: govet
+
+			syncWg.Add(1)
+
+			go func() {
+				defer syncWg.Done()
+
+				syncer.Run(syncCtx)
+			}()
+		}
+
+		if syncer != nil {
+			syncer.SetTimeServers(timeServers)
+		}
+
+		if syncDisabled {
+			timeSynced = true
+		}
+
+		if err = r.Update(ctx, time.NewStatus(), func(r resource.Resource) error {
+			r.(*time.Status).SetStatus(time.StatusSpec{
+				Epoch:        epoch,
+				Synced:       timeSynced,
+				SyncDisabled: syncDisabled,
+			})
+
+			return nil
+		}); err != nil {
+			return fmt.Errorf("error updating objects: %w", err) // nolint: govet
```

Comment:
```suggestion
			return fmt.Errorf("error updating objects: %w", err) //nolint:govet
```

The same everywhere. See https://github.com/golangci/golangci-lint/issues/1658 and https://github.com/golang/go/issues/43776

Reply:
yes, I missed that, `golangci-lint` was confusing me a lot on this line...